File: C++/ArrayString/28_3SumSmaller.cpp
#include <bits/stdc++.h>
using namespace std;

int countTriplets(int arr[], int n, int sum)
{
    sort(arr, arr + n);
    int ans = 0;
    for (int i = 0; i < n - 2; i++)
    {
        int j = i + 1, k = n - 1;
        while (j < k)
        {
            if (arr[i] + arr[j] + arr[k] >= sum)
                k--;

            else
            {
                ans += (k - j);
                j++;
            }
        }
    }
    return ans;
}

File: Java/ArrayString/28_3SumSmaller.java
public int twoSumSmaller(int[] arr,int si, int tar) {
        int ans = 0;
        int n = arr.length;
        for(int i=si;i<n;i++){
            for(int j=i+1;j<n;j++){
                int sum = arr[i] + arr[j];
                if(sum >= tar){
                    break;
                }
                ans++;
            }
        }
        return ans;
    }
    public int threeSumSmaller(int[] arr, int tar) {
        int n = arr.length;
        Arrays.sort(arr);
        int ans = 0;
        for(int i=0;i<n-2;i++){
            ans += twoSumSmaller(arr,i+1,tar-arr[i]);
        }
        return ans;
    }